src/OpenIddict.Client/OpenIddictClientOptions.cs

```diff
@@ -157,4 +157,11 @@ public sealed class OpenIddictClientOptions
     /// </summary>
     [EditorBrowsable(EditorBrowsableState.Advanced)]
     public HashSet<string> ResponseTypes { get; } = new(StringComparer.Ordinal);
+
+#if SUPPORTS_TIME_PROVIDER
+    /// <summary>
+    /// Gets the time provider
+    /// </summary>
+    public TimeProvider? TimeProvider { get; set; }
```

Comment:
Since a default value is always registered by the post-configuration initializer, it's better to remove the `?` here (otherwise, you have to use the null forgiving operator everywhere 😄)

```suggestion
    public TimeProvider TimeProvider { get; set; } = default!;
```

Reply:
Or maybe we should leave it nullable and do `TimeProvider?.GetUtcNow() ?? DateTimeOffset.UtcNow` where it's used? It would address the issue you had in the unit tests.